clients/feeder/feeder.go

```diff
@@ -447,3 +449,20 @@ func (c *Client) BlockTrace(ctx context.Context, blockHash string) (*starknet.Bl
 	}
 	return traces, nil
 }
+
+func findTargetDirectory(targetRelPath string) (string, error) {
+	wd, err := os.Getwd()
+	if err != nil {
+		return "", err
+	}
+
+	for root := wd; ; root = filepath.Dir(root) {
+		if _, err := os.Stat(filepath.Join(root, targetRelPath)); err == nil {
+			return filepath.Join(root, targetRelPath), nil
+		} else if !os.IsNotExist(err) {
+			return "", err
+		} else if newRoot := filepath.Dir(root); newRoot == root {
+			return "", os.ErrNotExist
+		}
+	}
+}
```

Comment:
I think the function can be simplified as follows:

```suggestion
func findTargetDirectory(targetRelPath string) (string, error) {
	root, err := os.Getwd()
	if err != nil {
		return "", err
	}

	for {
		targetPath := filepath.Join(root, targetRelPath)
		if _, err := os.Stat(targetPath); err == nil {
			return targetPath, nil
		} else if !os.IsNotExist(err) {
			return "", err
		}

		newRoot := filepath.Dir(root)
		if newRoot == root {
			return "", os.ErrNotExist
		}
		root = newRoot
	}
}
```